tests/repository_simulator.py

```diff
@@ -229,6 +239,8 @@ def _fetch_target(
 
         If hash is None, then consistent_snapshot is not used.
         """
+        self.fetch_tracker.targets.append((target_path, target_hash))
```

Comment:
We append to the fetch_tracker.targets all fetch requests. Should we append-only successful ones?